### ISSUE TITLE
Migrate MediaDeleteService to WorkManager

### ISSUE THREAD

**Issue by notandyvee** [wordpress-mobile/WordPress-Android#20991]
[Android API 34 foreground service policy changes](https://developer.android.com/about/versions/14/changes/fgs-types-required) were introduced. `UploadService ` should be migrated to `WorkManager`.

**Comment by dangermattic** [comment]
Thanks for reporting! 👍

**Comment by Akshaykomar890** [comment]
@nbradbury Can I work on this ?

**Comment by nbradbury** [comment]
@Akshaykomar890 Yes, please feel free to work on this, but note that this is a fairly involved task. An attempt at doing this for the upload service was made [here](https://github.com/wordpress-mobile/WordPress-Android/pull/19755) but never completed.

**Comment by Akshaykomar890** [comment]
@nbradbury Thanks for sharing, I believe this task requires more knowledge and experience, and as a beginner, I might need to look for smaller issues to work on. I'll continue exploring those and come back to this once I'm more comfortable.
